morepath/request.py

```diff
@@ -191,12 +199,12 @@ class (if this app doesn't handle them) to another app.
         if app is SAME_APP:
             app = self.app
 
-        info = app._get_deferred_mounted_path(obj)
+        info, app = app._get_deferred_mounted_path(obj)
```

Comment:
I was about to answer in support of your original argument when I realised what you mean. Yes, we actually use the app passed into the function to generate the link (that's the reason for this parameter anyway). So you originally convinced me and now you convince me again.

I would make awful candidate for president the way I flip-flop on this issue ;)